frontend/src/components/dialogs/SelectDialog.js

```diff
@@ -38,35 +39,80 @@ export default function SelectDialog({
   values,
   supportAdditionalInfo,
   className,
+  isSocial,
 }) {
   const classes = useStyles();
   const [element, setElement] = React.useState(null);
   const [additionalInfo, setAdditionalInfo] = React.useState({});
   const { locale } = useContext(UserContext);
   const texts = getTexts({ page: "general", locale: locale });
+  const orgTexts = getTexts({ page: "organization", locale: locale });
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+  const [textLabel, setTextLabel] = useState("");
 
   const handleClose = () => {
+    // restore defaults before closing
+    setElement(null);
+    setTextLabel("");
+    setHasError(false);
+    setErrorMessage("");
     onClose();
   };
 
   const applyElement = (event) => {
     event.preventDefault();
-    onClose(element, additionalInfo);
+
+    if (isSocial) {
+      const url = element.ask_for_full_website
+        ? additionalInfo[0].value
+        : element.base_url + additionalInfo[0];
+      if ("" !== verifySocialMediaLink(element, url, orgTexts)) {
+        setHasError(true);
+        setErrorMessage(verifySocialMediaLink(element, additionalInfo, orgTexts));
+      } else {
+        // restore defaults before closing
+        setHasError(false);
+        setErrorMessage("");
+        onClose(element, additionalInfo);
+        setElement(null);
+      }
+    } else {
+      // restore defaults before closing
+      onClose(element, additionalInfo);
+      setElement(null);
+    }
   };
 
   const handleSelectChange = (event) => {
-    const type = values.filter((x) => x.name === event.target.value)[0];
-    const typeAsRequired = {
-      key: type.key,
-      hide_get_involved: type.hide_get_involved,
-    };
-    setElement(typeAsRequired);
+
+    setHasError(false);
+    setErrorMessage("");
+
+    if (isSocial) {
+      setElement(values.filter((val) => val.name === event.target.value)[0]);
+    } else {
+       const type = values.filter((x) => x.name === event.target.value)[0];
+       const typeAsRequired = {
+           key: type.key,
+            hide_get_involved: type.hide_get_involved,
+       };
+       setElement(typeAsRequired);
+    }
+
     if (supportAdditionalInfo) {
       const value = values.filter((val) => val.name === event.target.value)[0];
-
-      if (value.additionalInfo.length > 0) {
+      if (isSocial) {
+        setTextLabel(
+          value?.ask_for_full_website ? value?.name : "Please enter your " + value?.name + " handle"
```

Comment:
1. We should use a translation file for this. I'd recommend creating a getSocialMediaTexts file. Also for the website I'd also put "Please enter your {channel} url", otherwise it's a bit confusing imo.

Reply:
File added with translations and suggested text.